### PR TITLE
Added basic documentation for select widget

### LINKED ISSUE #764: Add instruction on how to add options to `select` widget
At the moment https://www.netlifycms.org/docs/widgets/ doesn't cover how to add options to the `select` widget.

It would be good to have this documented and also mentioned about the ability to set distinct `labels` and `values`

```yaml
collections:
  foo:
    - {label: 'Option list', name: option-list', widget: select, options: [one, two,three]
    - {label: 'Option list', name: option-list', widget: select, options: [{label: 'Web Development', value: webdev]
```

https://github.com/netlify/netlify-cms/blob/master/src/components/Widgets/SelectControl.js#L18-L23

### PR DESCRIPTION
**Pull request by ackushiw**
**- Summary**
This helps people new to the platform with using the select widget and removes the need to go digging through code to work out the implementation. 

**- Test plan**
Used english dictionary

**- Description for the changelog**
closes #764

Added basic documentation for select widget

**- A picture of a cute animal (not mandatory but encouraged)**
'cute animal'

### CODE REVIEW

File: docs/widgets.md [hunk 1]
```diff
@@ -22,6 +22,23 @@ Widgets define the data type and interface for entry fields. Netlify CMS comes w
...
+collections:
+  - name: posts
+    label: Post
+    folder: "_posts"
+    slug: "{{year}}-{{month}}-{{day}}-{{slug}}"
+    create: true
+    fields:
+      - {label: Title, name: title, widget: string, tagname: h1}
+      - {label: Body, name: body, widget: markdown}
+      - {label: Align Content, name: align, widget: select, options: ['left', 'center', 'right']}
```
Comment (erquhart) [inline docs/widgets.md]: Closing #764 requires documentation of adding labels and values, would you mind adding an example of that, too? You could even trim this code sample down and just show the field config examples, e.g.:

```yaml
collections:
  - name: posts
     ...
     fields:
       ...
       - { label: Align Content, <rest of field> }
       - { label: Other example, <rest of field> }
```

**Comment by tech4him1** [comment]
@ackushiw I think it would also be nice if we note that you can use the `default` setting, to set the default option.

**Comment by ackushiw** [comment]
Yeah, I tried this in my own project but it didn't work. I assumed this
will be added in the next release, so didn't want to jump the gun. Let me
double check though.

**Comment by verythorough** [comment]
I'm going to merge this as an incremental improvement and keep #764 open for now.

**Pull request merged**